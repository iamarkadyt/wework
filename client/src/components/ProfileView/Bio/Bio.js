import React from 'react'
import { string } from 'prop-types'

const bio = ({
  bio
}) => bio ? (
  <section className="bio">
    <p className="header">About Me:</p>
    <p>{bio}</p>
  </section>
) : null

bio.propTypes = {
  bio: string.isRequired
}

export default bio
