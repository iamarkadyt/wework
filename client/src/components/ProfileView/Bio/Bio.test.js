import React from 'react'
import Bio from './Bio'
import { shallow } from 'enzyme'
import cloneDeep from 'lodash.clonedeep'

describe('Bio', () => {
  let mountedComponent, props

  const getMockProps = () => {
    return cloneDeep({
      bio: 'Some bio'
    })
  }

  const comp = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<Bio {...props} />)
    } 
    return mountedComponent
  }

  beforeEach(() => {
    mountedComponent = undefined
    props = getMockProps()
  })

  it('matches snapshot', () => {
    expect(comp()).toMatchSnapshot()
  })
})
