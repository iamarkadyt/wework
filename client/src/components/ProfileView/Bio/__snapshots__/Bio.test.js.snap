// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio matches snapshot 1`] = `
<section
  className="bio"
>
  <p
    className="header"
  >
    About Me:
  </p>
  <p>
    Some bio
  </p>
</section>
`;
